### ISSUE TITLE
Remove ChiseTest dependency

### ISSUE THREAD

**Issue by dobios** [chiselverify/chiselverify#51]
ChiselVerify is currently dependent on ChiselTest, which is no longer supported as of Chisel 6. 

Many features from ChiselTest have been integrated into the CIRCT compiler directly, which now has its own simulator `arcilator`. Using `arcilator` we can easily reimplement the `peek-poke-expect` interface that we rely on from ChiselTest.  This would allow ChiselVerify to survive past the deprecation of ChiselTest.

**Comment by hansemandse** [comment]
I agree that upgrading to the most recent version of Chisel should be in our main interests, but I can't tell how difficult this might be to do. Indeed, the `peek-poke-expect(-step)` logic should be easy to port, but the threading backend of ChiselTest is a bit more involved (albeit seemingly isolated to a few files [here](https://github.com/ucb-bar/chiseltest/tree/main/src/main/scala/chiseltest/internal)).

How do we approach this the best way?

**Comment by schoeberl** [comment]
AFAIK Kevin made ChiselTest still available for 6. We should publish ChiselVerify to support each major Chisel version. Then we shall work towards using arcilator. I have never tried that one. Maybe this is also a future solution for my second semester students with windows machines. Installing Verilator on Windows is a nightmare.
